jasminewd2/index.d.ts

```diff
@@ -44,6 +44,13 @@ declare namespace jasmine {
     toThrowError(expected?: new (...args: any[]) => Error | Promise<new (...args: any[]) => Error>, message?: string | RegExp | Promise<string | RegExp>): Promise<void>;
   }
 
+  interface ArrayLikeMatchers<T extends ArrayLike<T>> extends Matchers<T> {
```

Comment:
+1 to what @LOZORD said. This looks a bit suspicious. When you have ```T extends ArrayLike<T>```, then your constraint is essentially saying the type must be an ArrayLike<T> _in addition to_ still being the element type, ```T```. A regular array e.g. ```[1,2,3]``` will not meet this constraint. The only type that will is one that's an array of itself, like:

```ts
class SelfArray {
    length: number;
    [index: number]: SelfArray;
    someElementProperty: string
}
```

Is this what you want? If so, can you demonstrate it in a test, since it's rather subtle.